backend/post-service/src/main/java/com/speakapp/postservice/services/PostService.java

```diff
@@ -56,6 +56,33 @@ public PostGetDTO createPost(PostCreateDTO postCreateDTO, UUID userId) {
         );
     }
 
+    public PostGetDTO updatePost(PostCreateDTO postCreateDTO, UUID postId, UUID userId) {
+        Post postUpdated = postRepository.getPostByPostId(postId);
+
+        //TODO: Handling Exception when userId != postId
+
+        postMapper.updatePostFromPostCreateDTO(postCreateDTO, postUpdated);
+
+        postRepository.save(postUpdated);
```

Comment:
przy metodzie save jest zwracany aktualny obiekt, który stworzy sobie JPA, warto przypisać sobie ten obiekt do zmiennej i później ewentualnie na nim robić kolejne operacje i go ostatecznie zwrócić
powinieneś zrobić coś w stylu:
```
Post updatedPost = postRepository.save(postToUpdate);
```
A wcześniej nazwać post pobierany z bazy jako np. "postToUpdate" - zanim go zapiszesz (wywołasz save) to jest to tak naprawdę "post do aktualizacji"